addons/stock/models/stock_move_line.py

```diff
@@ -572,8 +572,12 @@ def _action_done(self):
 
         if ml_ids_tracked_without_lot:
             mls_tracked_without_lot = self.env['stock.move.line'].browse(ml_ids_tracked_without_lot)
-            raise UserError(_('You need to supply a Lot/Serial Number for product: \n - ') +
-                              '\n - '.join(mls_tracked_without_lot.mapped('product_id.display_name')))
+            raise UserError(
+                _(
+                    "You need to supply a Lot/Serial Number for product: \n - %(products)s",
+                    products="\n - ".join(mls_tracked_without_lot.mapped("product_id.display_name")),
```

Comment:
Same suggestion as earlier, try to put the dash for the first element of the list outside the translated string, so you can be sure that the translators won't change it, which would make the first element of the list different from the others